Enhance: add ability to set title for floating/dockable map window

SlySven opened the pull request:
This provides the same sort of feature to the map widget as was applied to user windows in PR https://github.com/Mudlet/Mudlet/pull/3585.

Obviously it only works for the floating/dockable type mapper - and it will tell the user this if they try it on the embedded one.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Enhance: add ability to set title for floating/dockable map window

This provides the same sort of feature to the map widget as was applied to
user windows in PR #3585.

Obviously it only works for the floating/dockable type mapper - and it
will tell the user this if they try it on the embedded one.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (approved):
Could we have a reset too 😉

SlySven commented:
Same process as the `setUserWindowTitle(...)` one, call it with no argument or with an empty string...

vadi2 commented:
That one does have a reset function. Gotta keep the API not weird for the non-programmers among us.

SlySven commented:
Ah, that is done in an external Lua wrapper, isn't it. Whilst I could probably come up with a wrapper to provide it for the naked mapper case - I am not up to delving into any Geyser code. If I do the former can you recommend someone who knows more about that, maybe, @demonnic ...?

SlySven committed:
Revise: add resetMapWindowTitle() as requested to GUIUtils.lua

Note that something may be needed to accommodate setting/resetting the
map window widget's title in Geyser - but I am not sure I am up to that.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Okay - I have done what ***I*** can...

vadi2 commented:
@Edru2 is this something you could add?

Edru2 committed:
Update GeyserMapper.lua

Edru2 commented:
Check out the PR at https://github.com/SlySven/Mudlet/pull/9
Which adds the Geyser functionality.
